### ISSUE TITLE
Cannot unzip files on server

### ISSUE THREAD

**Issue by KenEucker** [ZJONSSON/node-unzipper#226]
I can use this package during local development without issue, however, when I deploy my application to the server it doesn't seem to be parsing the zip files at all. ( it does not enter the 'entry' event, nor does it enter 'error' ).

Can someone help point me in the right direction here, I feel like it's some missing dependency of some kind but I have all peers installed, to my knowledge, and I don't see any warnings when running `npm install`.

This is the error I am seeing on the server:
```
node:internal/process/promises:225
          triggerUncaughtException(err, true /* fromPromise */);
```

My code works on my machine without issue. Server environment is Ubuntu 20.04 LTS. 

Thanks in advance for any help you can provide.

**Comment by KenEucker** [comment]
Following up on this, I think I have one more thing to add:

I can get unzipper to work successfully when running PM2 on the server, however, if I attempt to unzip files when running my node application directly the file does not appear to unpack any items. This seems like it would be a permission issue, but what is weird is that the process that creates the archive in the first place is the one that immediately attempts to (and successfully under PM2) unpack it. I would expect it to have the same permissions to the file.

**Comment by KenEucker** [comment]
Aaaaaannnnnddddd I figured out what I was doing wrong:

I was trying to use the result of the following as a promise, that I could `await`.

```
await createReadStream(tempTemplateArchive)
	.pipe(unzipper.Parse())
        .on('entry', (entry) => {
        ...
```

When, in your README, you have an example showing how to properly await each entry event. I changed my code to the following:
```
const tempArchiveFiles = createReadStream(tempTemplateArchive).pipe(unzipper.Parse({forceStream: true}))
							
for await (const entry of tempArchiveFiles) {
...
```

And now it's working just fine. The issue was that the application, when running in memory on the server, would exit before finishing the unpacking. I don't know why this works on my local machine and not the server, but it was an issue with my expectations and not paying closer attention to the README.

Thank you for your hard work and for your contributions to FOSS!
